### DISCUSSION TITLE
Connecting to [My IP] Indefinitely

### DISCUSSION THREAD

**Discussion by BahnBehringer** [Pryaxis/TShock#1696] [General]
* TShock version: 4.3.26
* TShock build number (if known): Unknown

#### Reproduction steps (if applicable)?

1. Install Tshock version 4.3.26
2. extract all files
3. run the server
4. provide IP and port to friends once server setup is complete
5. Watch helplessly as they still can't connect despite a port scanner successfully connecting to the server during troubleshooting.

#### Any stack traces or error messages (if known)? N/A

No error messages are showing. The error is that no one can join my server despite the port being open and the IP being correctly provided. The server doesn't even show a connection attempt. However, when I use a port scanner on a website to see if the port is open, the server registers the port scan as a connection attempt and the port is identified as open on the scanner.

#### Any screenshots? Nothing to screenshot because there is no error

!!!!!! PLEASE FILL IN THE TEMPLATE ABOVE THANK YOU VERY MUCH !!!!!! np <3

**Comment by hakusaro** [comment]
Hi! Can you provide some information? Specifically:

* What IP are you giving the online port scanner? Please redact if you'd like by giving us the first two octets: `72.39.xxx.xxx`.
* What IP are you giving the friends?
* Have you already completed port forwarding steps? If so, can you screenshot your router's config?
* What port are you running the server on?

_I have converted this issue from a GitHub issue into a GitHub Discussion. This enables Q&A systems, upvoting, etc._

**Reply by BahnBehringer** [reply]
I'll explain from the beginning. I downloaded Tshock about 5 days ago for the first time. I forwarded my port, started the server, provided my friends the IP address and off we went and it was great. However, the very next morning I started up the Tshock server with the same parameters, on the same world, and my friends can't connect. It simply says: "Connecting to [My IP]" Indefinitely. I used a port scanner when the server was open and my port was confirmed open and the server showed the connection attempt. However, when my friends attempt to join the server doesn't even show a connection attempt. The IP I provided my friends and the port scanner were the same: 73.34.xxx.xxx. My port forwarding was successful and is still in effect. Finally, I am running the server on port 7777. I can't screenshot my router's config because my ISP forces port forwards to be done via their website and the router config simply provides a link to their website. My apologies.

**Reply by hakusaro** [reply]
Are you still able to connect locally? I'm so sorry for the delay on this -- my life is kind of chaotic right now. We've seen issues where people sometimes have the server crash due or get into an inconsistent state. You can rule this out if you can connect on a local IP from Terraria while users normally get stuck connecting.

**Reply by BahnBehringer** [reply]
I am still able to connect locally. Wireshark shows me a bunch of LLMNR standard query's when my friend tries to join but the server still shows no connection attempts. I disabled my router firewall to no avail.

**Comment by moisterrific** [comment]
If you can still connect to the server and your friends cannot it’s a port forwarding issue on your end and not a TShock issue. 

Perhaps the port got messed up after you restarted your PC or router? I would first try double checking if the port(s) are forwarded properly.

**Reply by BahnBehringer** [reply]
It is not a port forwarding issue. I have confirmed the port is open via an external website.

**Reply by moisterrific** [reply]
Can you confirm if you can connect to the server locally?

**Reply by BahnBehringer** [reply]
I can connect locally.

**Reply by BahnBehringer** [reply]
Wireshark shows me a bunch of LLMNR standard query's when my friend tries to join but the server still shows no connection attempts. I disabled my router firewall to no avail.

**Reply by moisterrific** [reply]
Then it’s extremely likely a port forwarding issue, not TShock. I’ve had this same problem a couple times in the past when my router settings were reset due to power outage.

Double check your router and Windows firewall settings to see if they could be denying connections.

**Reply by BahnBehringer** [reply]
I do not believe it is a port forwarding issue due to the troubleshooting conducted. However, how would you recommend fixing the issue? remove the port forward and then add it again? Xfinity (my ISP) said they had no ability to change any firewall access to my network and that the router firewall should be the only thing blocking it.

**Reply by moisterrific** [reply]
Could also be Windows firewall. When I was setting up my server for the first time it was a router firewall issue. Then later on it turned out to be Windows firewall blocking connections due to the enterprise configuration on my computer.

**Reply by BahnBehringer** [reply]
windows firewall says that terrariaserver.exe is allowed privately and publically

**Reply by moisterrific** [reply]
Try double checking your router settings, make sure everything is set up properly. Here are my settings for reference if it helps: 
![D7DBC29C-8AD4-4053-85CC-C730945AA7DD](https://user-images.githubusercontent.com/57187883/80546157-f1944000-8982-11ea-8433-db58e4f1e797.jpeg)

**Reply by BahnBehringer** [reply]
Xfinity doesn't offer the ability of doing port forwards via the router. They removed it within the past year and replaced it with an html-based version through their website. I have enabled the port forward for TCP/UDP on port 7777 (same port server is using) my external IP address is also shown on their website and is confirmed on whatismyipaddress.com. The IP address is the correct one being used by my friend to join as well as the port. A man from the discord of Tshock was able to connect (but only when using a VPN)

**Reply by moisterrific** [reply]
Still doesn’t seem to be a TShock issue since local connections still work. If it was then nobody at all can connect.

**Reply by moisterrific** [reply]
Maybe get a better ISP that give you full control of router settings lol

**Reply by BahnBehringer** [reply]
I'm currently on the phone with them trying to figure it out. Thank you for your time!

**Reply by hakusaro** [reply]
@moisterrific in the US, basically every ISP has a monopoly. Xfinity or bust is normal :/

**Comment by kubedzero** [comment]
I also have this issue, I run TShock on my LAN but also have a port forward set up for known guests to join. Random IP addresses (NOT my guests, but malicious/bot IPs)  get marked as `connecting` in the CMD window and then when I connect from the LAN the Terraria game notes that I'm stuck on "Server Found." It appears to connect partially but hang on the login/password prompt. 

To unstick it, I've found pressing Enter a few times on the CMD console will trigger a save (I guess this is the save on disconnect function but am not positive) and then my local Terraria game makes it to the password prompt. 

I've just hopefully disabled the save on disconnect option today by changing `SaveWorldOnLastPlayerExit` and am hoping that fixes the issue, but I'll report back soon enough if I see continued connection issues.

**Reply by moisterrific** [reply]
I’ve had issues like this occasionally, but I was mostly using an unofficial mobile version. 

Not sure if the desktop server also had the same problem.
